File: php/features.php
<?php
declare(strict_types=1);

// IpAddress SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class IpAddressFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new IpAddressBaseFeature();
            case "test":
                return new IpAddressTestFeature();
            default:
                return new IpAddressBaseFeature();
        }
    }
}

File: rb/features.rb
# IpAddress SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module IpAddressFeatures
  def self.make_feature(name)
    case name
    when "base"
      IpAddressBaseFeature.new
    when "test"
      IpAddressTestFeature.new
    else
      IpAddressBaseFeature.new
    end
  end
end
